data/src/test/java/org/apache/iceberg/io/TestWriterMetrics.java

```diff
@@ -378,4 +351,23 @@ public void testMaxColumnsWithDefaultOverride() throws IOException {
           .isEqualTo(1);
     }
   }
+
+  protected void checkRowStatistics(Map<Integer, ByteBuffer> bounds) {
+    assertThat(bounds).hasSize(4);
+    assertThat((int) Conversions.fromByteBuffer(Types.IntegerType.get(), bounds.get(1)))
+        .isEqualTo(3);
+    assertThat(bounds).doesNotContainKey(2);
+    assertThat(bounds).doesNotContainKey(3);
+    assertThat(bounds).doesNotContainKey(4);
+    assertThat((long) Conversions.fromByteBuffer(Types.LongType.get(), bounds.get(5)))
+        .isEqualTo(3L);
+  }
+
+  protected void checkNotExistingRowStatistics(Map<Integer, ByteBuffer> bounds) {
+    assertThat(bounds).hasSize(2);
+    assertThat((int) Conversions.fromByteBuffer(Types.IntegerType.get(), bounds.get(1)))
+        .isEqualTo(3);
+    assertThat((long) Conversions.fromByteBuffer(Types.LongType.get(), bounds.get(5)))
+        .isEqualTo(3L);
+  }
```

Comment:
We need to do different checks for the `FileWriterFactory` implementations which has deprecated the position deletes where the rows are embedded